### ISSUE TITLE
Physics interpolation seems to break Sprite2D

### ISSUE THREAD

**Issue by AlexOtsuka** [godotengine/godot#93586]
### Tested versions

Reproducible in both 4.3 betas and current master.

### System information

Godot v4.3.dev (4a0160241) - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1070 (NVIDIA; 31.0.15.3623) - Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz (8 Threads)

### Issue description

It seems like a number of scenarios break `Sprite2D` at runtime when enabling physics interpolation. `skew` seems to be ignored entirely, `rotation` and `scale` work by themselves, but not when used together with a non-proportional scale value.

### Steps to reproduce

In the screenshots below,

- Column 1 is untouched, default `Sprite2D` properties.
- Column 2 is using `skew` only.
- Column 3 is using `rotation` only.
- Column 4 is using `rotation` and a non-proportional `scale` value.

Besides that, it's a fresh empty project without a single line of code (MRP included below).

Setup inside the editor (looks as expected):

![in editor](https://github.com/godotengine/godot/assets/109620202/60e70991-ccd5-4f20-be89-7c519046fcb8)

At runtime, physics interpolation OFF (looks as expected):

![no interpolation](https://github.com/godotengine/godot/assets/109620202/b9d37170-7051-46ba-8e4a-487fce6d7308)

At runtime, physics interpolation ON (broken):

![interpolation on](https://github.com/godotengine/godot/assets/109620202/d1b01835-5a3d-4add-b90a-17574d73e5e0)

### Minimal reproduction project (MRP)

[SpriteBugMRP.zip](https://github.com/user-attachments/files/15965548/SpriteBugMRP.zip)

Enable physics interpolation on/off in `Project Settings > Physics > Common (Advanced Settings)`. Note that you might have to run the project, close it, and run it again every time you change the setting for it to take effect.

**Comment by akien-mga** [comment]
CC @rburing @lawnjelly

**Comment by cosparks** [comment]
I checked out the mrp and this issue seems to be caused by that fact that transform interpolation isn't implemented correctly. The following unit test fails (values are copied from the transform of row 2, column 2 in the mrp):

```c++
TEST_CASE("[TransformInterpolator] Skewed Transform -- prev == cur") {
	Transform2D skewed = Transform2D(Vector2(1.f, 0.f), Vector2(-0.766044676f,0.642787337f), Vector2(0.f, 0.f));
    Transform2D result = Transform2D();
	TransformInterpolator::interpolate_transform_2d(skewed, skewed, result, 0.f);
    CHECK(result == skewed);
}
```
```
ERROR: CHECK( result == skewed ) is NOT correct!
values: CHECK( [X: (1, 0), Y: (0, 1), O: (0, 0)] == [X: (1, 0), Y: (-0.766045, 0.642787), O: (0, 0)] )
```

I know the test shouldn't be checking for equality because of floating point error, but clearly the second column of the result transform is incorrect.

**Comment by cosparks** [comment]
Interpolating the transform's skew value appears to have fixed the results in the second column:
![image](https://github.com/godotengine/godot/assets/20360622/2b445ea5-4bb6-4c99-909c-a3e9c4de8104)

**Comment by lawnjelly** [comment]
Ah sorry I missed this, can reproduce in 3.x.

Great issue, the order of operations is likely the wrong way around in the interpolator (a while since I did this, but I suspect copy pasted from elsewhere in the engine). It may not have been reported before in 3.x because we don't have a skew and it is rarely used.

Will see if I can fix this up before 3.6 RC 1. :+1:

**Comment by cosparks** [comment]
If `Transform2D::interpolate_with` exists then this should definitely use it. No sense having the exact same logic in two places.

I'd be happy to put up the PR for this
